src/main/java/com/exasol/dbbuilder/dialects/exasol/ConnectionDefinition.java

```diff
@@ -93,12 +91,16 @@ public boolean hasParent() {
 
     @Override
     public DatabaseObject getParent() {
-        throw new DatabaseObjectException(this, ExaError.messageBuilder("E-TDBJ-8").message("Illegal attempt to access parent object of a CONNECTION which is a top-level object.").toString());
+        throw new DatabaseObjectException(this,
+                ExaError.messageBuilder("E-TDBJ-8")
+                        .message("Illegal attempt to access parent object of a CONNECTION which is a top-level object.")
+                        .toString());
     }
 
     @Override
     // [impl->dsn~dropping-connections~1]
-    public void drop() {
+    protected void dropInternally() {
+        verifyNotDeleted();
```

Comment:
As far as I can understand the super class method `drop()` should already call `verifyNotDeleted()`, right?
```suggestion
    protected void dropInternally() {
```